x/programs/examples/simulator/cmd/program.go

```diff
@@ -0,0 +1,307 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package cmd
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/spf13/cobra"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/utils"
+	"github.com/ava-labs/hypersdk/x/programs/examples/imports/program"
+	"github.com/ava-labs/hypersdk/x/programs/examples/imports/pstate"
+	"github.com/ava-labs/hypersdk/x/programs/examples/storage"
+	"github.com/ava-labs/hypersdk/x/programs/runtime"
+)
+
+const (
+	inheritIDKey = "inherit"
+)
+
+var (
+	programID ids.ID
+)
+
+type Program struct {
+	Name        string   `yaml:"name"`
+	Description string   `yaml:"description"`
+	Actions     []Action `yaml:"actions"`
+	CallerKey   string   `yaml:"caller_key"`
+	Config      Config   `yaml:"config"`
+}
+
+type Config struct {
+	// Maximum number of pages of memory that can be used.
+	// Each page represents 64KiB of memory.
+	MaxMemoryPages uint64 `yaml:"max_memory_pages"`
+}
+
+type Action struct {
```

Comment:
We probably want to call this something else to avoid conflating this with `tx.Action`.

I'm thinking `step` may work?